### PR TITLE
feat: user management api

### PR DESCRIPTION
**Pull request by sevu1**
Issue #, if available:

Description of changes: user CRUD api

Checklist:

* [X] Have you successfully deployed to an AWS account with your changes?
* [X] Have you written new tests for your core changes, as applicable?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: solutions/swb-app/src/userRoutes.ts [hunk 1]
```diff
@@ -7,39 +7,153 @@
...
-export function setUpUserRoutes(router: Router, user: UserManagementService): void {
+export function setUpUserRoutes(router: Router, userService: UserManagementService): void {
   router.post(
     '/users',
     wrapAsync(async (req: Request, res: Response) => {
-      processValidatorResult(validate(req.body, CreateUserSchema));
-      const response = await user.createUser(req.body);
-      res.status(201).send(response);
+      try {
+        processValidatorResult(validate(req.body, CreateUserSchema));
```
Comment (FernandoAranda) [inline solutions/swb-app/src/userRoutes.ts:33]: not sure if its out of the scope but can we use zod instead of schema

Comment (sevu1) [reply solutions/swb-app/src/userRoutes.ts:33]: Yea, I'd prefer not to increase scope of this PR, so only zod parser for new code was included. This should be considered as a separate work item

**Review by FernandoAranda — APPROVED** [review]

File: solutions/swb-app/src/userRoutes.ts [hunk 2]
```diff
@@ -7,39 +7,153 @@
...
+
+        if (isUserAlreadyExistsError(err)) {
+          throw Boom.conflict(`User with this email already exist.`);
+        }
+
+        if (Boom.isBoom(err)) {
+          throw err;
+        }
+
+        throw Boom.badImplementation(`Could not find create user`);
```
Comment (bhudnell) [inline solutions/swb-app/src/userRoutes.ts:49]: maybe something like "Could not create user" would be better. The current message isnt clear to me.

**Review by bhudnell — APPROVED** [review]

File: solutions/swb-app/src/userRoutes.ts (hunk 2 above)
Comment (sevu1) [reply solutions/swb-app/src/userRoutes.ts]: Copy-paste issue  , neither clear to me  😄

**Review by bhudnell — APPROVED** [review]

**Review by FernandoAranda — APPROVED** [review]

**Pull request merged**